pkg/registry/pod/strategy.go

```diff
@@ -464,12 +464,12 @@ func streamLocation(
 		return nil, nil, err
 	}
 	loc := &url.URL{
-		Scheme:   nodeScheme,
-		Host:     fmt.Sprintf("%s:%d", nodeHost, nodePort),
+		Scheme:   connectionInfo.Scheme,
+		Host:     fmt.Sprintf("%s:%d", connectionInfo.Hostname, connectionInfo.Port),
```

Comment:
@liggitt `connectionInfo.Port` is `uint`, while `JoinHostPort` wants a string, so sprintfs conversion would be required anyway, should we still use it?